### PR TITLE
Merge the student group and student batch

### PR DESCRIPTION
**Pull request by manassolanki**
- Change the student group doctype
- Delete the student batch doctype
- Patch to map the student batch data on the student group
- Changes in the linked doctype
- Cleanup for deprecated doctype

### CODE REVIEW

**Review by nabinhait — CHANGES_REQUESTED** [review]
Screenshot please

File: erpnext/patches/v8_0/merge_student_batch_and_student_group.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+def execute():
+
+	# for converting student batch into student group
+	frappe.reload_doctype("Student Group")
+	student_batches = frappe.db.sql('''select name as student_group_name, student_batch_name as batch, program, academic_year, academic_term
+		from `tabStudent Batch`''', as_dict=1)
+	for student_batch in student_batches:
+		student_batch.update({"doctype":"Student Group", "group_based_on": "Batch"})
+		doc = frappe.get_doc(student_batch)
+		doc.save()
```
Comment (nabinhait) [inline erpnext/patches/v8_0/merge_student_batch_and_student_group.py]: Why saving doc here? you can save after adding child records.

File: erpnext/patches/v8_0/merge_student_batch_and_student_group.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import frappe
+from frappe.model.utils.rename_field import *
+from frappe.model.mapper import get_mapped_doc
+
+
+def execute():
+
+	# for converting student batch into student group
+	frappe.reload_doctype("Student Group")
+	student_batches = frappe.db.sql('''select name as student_group_name, student_batch_name as batch, program, academic_year, academic_term
```
Comment (nabinhait) [inline erpnext/patches/v8_0/merge_student_batch_and_student_group.py]: use multilines after 80 chars

File: erpnext/patches/v8_0/merge_student_batch_and_student_group.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+			doc.extend("students", student_list)
+
+		instructor_list = frappe.db.sql('''select instructor, instructor_name from `tabStudent Batch Instructor` where parent=%s''', (doc.name), as_dict=1)
+		if instructor_list:
+			doc.extend("instructors", instructor_list)
+		doc.save()
+
+	# delete the student batch and child-table
+	frappe.delete_doc("DocType", "Student Batch", force=1)
+	# frappe.db.sql("drop table if exists `tabStudent Batch`")
```
Comment (nabinhait) [inline erpnext/patches/v8_0/merge_student_batch_and_student_group.py]: remove commented lines

File: erpnext/schools/doctype/course_schedule/course_schedule.py [hunk 4]
```diff
@@ -11,51 +11,36 @@ class CourseSchedule(Document):
...
 		"""Set document Title"""
 		self.title = self.course + " by " + (self.instructor_name if self.instructor_name else self.instructor)
-
-	def validate_mandatory(self):
-		if not (self.student_batch or self.student_group):
-			frappe.throw(_("""Student Batch or Student Group is mandatory"""))
 	
 	def validate_course(self):
-		if self.student_group:
+		if self.student_group and frappe.db.get_value("Student Group", self.student_group, "group_based_on") == "Course":
```
Comment (nabinhait) [inline erpnext/schools/doctype/course_schedule/course_schedule.py]: Why 2 queries? It should be like this:

```
if self.student_group:
    group_based_on, course = frappe.db.get_value("Student Group", self.student_group, ["group_based_on", "course"])
    if group_based_on == "Course":
        self.course = course
```

File: erpnext/schools/doctype/course_scheduling_tool/course_scheduling_tool.py [hunk 5]
```diff
@@ -22,8 +22,8 @@ def schedule_course(self):
 		self.validate_mandatory()
 		self.validate_date()
 		self.instructor_name= frappe.db.get_value("Instructor", self.instructor, "instructor_name")
-		
-		if self.student_group:
+
+		if self.student_group and frappe.db.get_value("Student Group", self.student_group, "group_based_on") == "Course":
```
Comment (nabinhait) [inline erpnext/schools/doctype/course_scheduling_tool/course_scheduling_tool.py]: same as above

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/schools/doctype/course_schedule/course_schedule.py (hunk 4 above)
Comment (manassolanki) [inline erpnext/schools/doctype/course_schedule/course_schedule.py]: Don't know this method

**Pull request merged**